src/main/java/fr/insalyon/creatis/gasw/executor/batch/internals/BatchJob.java

```diff
@@ -0,0 +1,187 @@
+package fr.insalyon.creatis.gasw.executor.batch.internals;
+
+import fr.insalyon.creatis.gasw.GaswException;
+import fr.insalyon.creatis.gasw.execution.GaswStatus;
+import fr.insalyon.creatis.gasw.executor.batch.config.json.properties.BatchEngines;
+import fr.insalyon.creatis.gasw.executor.batch.internals.commands.RemoteCommand;
+import fr.insalyon.creatis.gasw.executor.batch.internals.commands.items.Cat;
+import fr.insalyon.creatis.gasw.executor.batch.internals.terminal.RemoteFile;
+import fr.insalyon.creatis.gasw.executor.batch.internals.terminal.RemoteOutput;
+import fr.insalyon.creatis.gasw.executor.batch.internals.terminal.RemoteTerminal;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.extern.log4j.Log4j;
+
+@Log4j
+@RequiredArgsConstructor
+@Setter
+public class BatchJob {
+
+    @Getter
+    final private BatchJobData data;
+
+    @Getter
+    private boolean     terminated = false;
+    private GaswStatus  status = GaswStatus.NOT_SUBMITTED;
+
+    private void createBatchFile() throws GaswException {
+        final BatchFile batchFile = new BatchFile(data);
+        final RemoteOutput output;
+
+        output = new RemoteTerminal(data.getConfig()).oneCommand("echo -en '" + batchFile.build().toString() + "' > "
+                + data.getWorkingDir() + data.getJobID() + ".batch");
+
+        if (output == null || output.getExitCode() != 0 || ! output.getStderr().getContent().isEmpty()) {
+            throw new GaswException("Impossible to create the batch file");
+        }
+    }
+
+    /**
+     * Upload all the data to the job directory.
+     * 
+     * @throws GaswException
+     */
+    public void prepare() throws GaswException {
+        final RemoteTerminal rt = new RemoteTerminal(data.getConfig());
+
+        rt.connect();
+        for (final RemoteFile file : data.getFilesUpload()) {
+            log.info("Uploading file : " + file.getSource());
+            rt.upload(file.getSource(), file.getDest());
+        }
+        rt.disconnect();
+    }
+
+    /**
+     * Download all the data created by the jobs (not the app output) but the logs.
+     */
+    public void download() {
+        final RemoteTerminal rt = new RemoteTerminal(data.getConfig());
+
+        try {
+            rt.connect();
+            for (final RemoteFile file : data.getFilesDownload()) {
+                log.info("Downloading file : " + file.getSource());
+                rt.download(file.getSource(), file.getDest());
+            }
+            rt.disconnect();
+
+        } catch (GaswException e) {
+            log.error("Failed to download the files !", e);
```

Comment:
See if it is ok to silence this exception.
I think it should be silenced in the calling method if necessary.